src/rotation_types.jl

```diff
@@ -0,0 +1,192 @@
+
+
+# a default element type
+macro DefaultElType(); Float64; end  # need this to be a macro for type stability in type_methods.jl promote_type_sp functions
+
+# some trait style stuff
+numel{T}(::Type{T}) = length(fieldnames(T))    # function to get the number of elements in each parameterization
+n_params{T}(::Type{T}) = length(T.parameters)  # the number of template parameters
+
+###################################################
+# Rotation matrix (typeofalias of Mat{3,3,Float64})
+# N.B. all other parameterizations T should implement
+# a convert_rotation(T, RotMatrix) method
+###################################################
+
+@doc """
+A type alias for rotation matrices (alias of Mat{3,3,T})
+""" ->
+typealias RotMatrix{T <: Real}   Mat{3,3,T}
+#immutable RotMatrix{T <: AbstractFloat} <: FixedArray{T,2,Tuple{3,3}} # this would be better but much more effort without an implicit conversion
+#    _::Tuple{Tuple{T, T, T}, 
+#             Tuple{T, T, T},
+#             Tuple{T, T, T}}
+#end
+
+ 
+# fix weirdness, try: mat = Mat(eye(3)); @code_warntype convert(typeof(mat), mat)  
```

Comment:
True; but until they've made it into a tagged version of FixedSizeArrays we're stuck with it.